### PR TITLE
Feature/adopt typescript

### PR DESCRIPTION
**Pull request by sfryffdc**

### CODE REVIEW

File: src/openid/shared/openIdClient.d.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+declare module 'openid-client' {
```
Comment (dfruddffdc) [inline src/openid/shared/openIdClient.d.ts]: Don't type definition files need copyright notices too?

File: test/openid/shared/envValidator.test.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+			expect(() => validate(env)).to.throw('Invalid parameter: openidIssuerURI cannot be empty.');
+
+		});
+
+	});
+
+	it('should resolve with env if env is ok', () => {
+
+		// given - when - then
+		expect(validate(env)).to.deep.equal(env);
```
Comment (dfruddffdc) [inline test/openid/shared/envValidator.test.ts]: I generally use `eql`, which is equivalent to `deep.equal` - but live and let live I say

File: test/openid/shared/userinfo.test.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+		it('should throw an error on failure', async () => {
+
+			// Given
+
+			sinon.stub(issuer, 'constructIssuerClient').rejects(new Error());
+
+			// When - Then
+
+			expect(userinfo.getUserInfo(env, 'ABCDE123'))
```
Comment (dfruddffdc) [inline test/openid/shared/userinfo.test.ts]: Should you await the rejection? Is there a race condition on this test?

Comment (sfryffdc) [reply test/openid/shared/userinfo.test.ts]: Yes and yes there was. Fixed.

File: package.json [hunk 4]
```diff
@@ -1,57 +1,81 @@
...
-    "chai": "^4.1.2",
-    "chai-as-promised": "^7.1.1",
-    "jsdoc": "^3.5.5",
-    "mocha": "^5.2.0",
-    "nyc": "^12.0.2",
-    "proxyquire": "^2.0.1",
-    "sinon": "^5.1.0"
-  }
+	"name": "@financialforcedev/orizuru-auth",
+	"version": "3.0.4",
```
[the remark this answers is not in the document]
Comment (sfryffdc) [reply package.json]: True. Fixed.

**Pull request merged**